qutip/core/tensor.py

```diff
@@ -77,53 +77,36 @@ def tensor(*args):
      [ 0.+0.j  1.+0.j  0.+0.j  0.+0.j]
      [ 1.+0.j  0.+0.j  0.+0.j  0.+0.j]]
     """
-
     if not args:
         raise TypeError("Requires at least one input argument")
-
-    if len(args) == 1 and isinstance(args[0], (list, np.ndarray)):
-        # this is the case when tensor is called on the form:
-        # tensor([q1, q2, q3, ...])
-        qlist = args[0]
-
-    elif len(args) == 1 and isinstance(args[0], Qobj):
-        # tensor is called with a single Qobj as an argument, do nothing
-        return args[0]
-
-    else:
-        # this is the case when tensor is called on the form:
-        # tensor(q1, q2, q3, ...)
-        qlist = args
-
-    if not all([isinstance(q, Qobj) for q in qlist]):
-        # raise error if one of the inputs is not a quantum object
-        raise TypeError("One of inputs is not a quantum object")
-
-    out = Qobj()
-
-    if qlist[0].issuper:
-        out.superrep = qlist[0].superrep
-        if not all([q.superrep == out.superrep for q in qlist]):
-            raise TypeError("In tensor products of superroperators, all must" +
-                            "have the same representation")
-
-    out.isherm = True
-    for n, q in enumerate(qlist):
-        if n == 0:
-            out.data = q.data
-            out.dims = q.dims
-        else:
-            out.data  = zcsr_kron(out.data, q.data)
-            
-            out.dims = [out.dims[0] + q.dims[0], out.dims[1] + q.dims[1]]
-
-        out.isherm = out.isherm and q.isherm
-
-    if not out.isherm:
-        out._isherm = None
-
-    return out.tidyup() if qutip.settings.auto_tidyup else out
-
+    if len(args) == 1 and isinstance(args[0], Qobj):
+        return args[0].copy()
+    if len(args) == 1:
+        try:
+            args = tuple(args[0])
+        except TypeError:
+            raise TypeError("requires Qobj operands") from None
+    if not all(isinstance(q, Qobj) for q in args):
+        raise TypeError("requires Qobj operands")
+    if not all(q.superrep == args[0].superrep for q in args[1:]):
+        raise TypeError("".join([
+            "In tensor products of superroperators,",
```

Comment:
Seems like an elaborate way to concatenate strings

Reply:
Honestly it's just a bit of force-of-habit these days.  The `"".join()` idiom is the easiest to format in PEP8 and technically it has the best asymptotic complexity of all the string concatenation methods, as irrelevant as that is for small numbers of strings which are called exactly once at a failure point anyway...

In this particular case, where there's no formatting to be done, the simplest method doesn't even need any operators, since string literals are automatically concatenated as part of the parsing.
```
raise TypeError(
    "In tensor products of superoperators,"
    " all must have the same representation"
)
```